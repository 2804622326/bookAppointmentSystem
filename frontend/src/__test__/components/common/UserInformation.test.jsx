import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import UserInformation from '../../../components/common/UserInformation.jsx';

const appointment = {
  appointmentNo: 'A1',
  patient: { firstName: 'Jane', lastName: 'D', email: 'j@d.com', phoneNumber: '1' },
  veterinarian: { firstName: 'Sam', lastName: 'Vet', email: 's@v.com', phoneNumber: '2', specialization: 'Surg' },
};

test('shows patient information for vet user', () => {
  const { asFragment } = render(
    <UserInformation userType="VET" appointment={appointment} />
  );
  expect(screen.getByText(/jane/i)).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});

test('shows vet information for patient user', () => {
  const { asFragment } = render(
    <UserInformation userType="PATIENT" appointment={appointment} />
  );
  expect(screen.getByText(/dr\./i)).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});
