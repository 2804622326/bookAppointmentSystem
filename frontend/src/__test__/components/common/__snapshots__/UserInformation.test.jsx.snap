// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows patient information for vet user 1`] = `
<DocumentFragment>
  <div
    class="mt-2 mb-2"
    style="background-color: whitesmoke;"
  >
    <h5>
      Patient  Information
    </h5>
    <p
      class="text-info"
    >
      Appointment No: A1
    </p>
    <p>
      Name: Jane D
    </p>
    <p>
      Email: j@d.com
    </p>
    <p
      class="text-info"
    >
      Phone Number: 1
    </p>
  </div>
</DocumentFragment>
`;

exports[`shows vet information for patient user 1`] = `
<DocumentFragment>
  <div
    class="mt-2 mb-2"
    style="background-color: whitesmoke;"
  >
    <h5>
      Veterinarian  Information
    </h5>
    <p
      class="text-info"
    >
      Appointment No: A1
    </p>
    <p>
      Name: Dr. Sam Vet
    </p>
    <p
      class="text-info"
    >
      Specialization: Surg
    </p>
    <p>
      Email: s@v.com
    </p>
    <p
      class="text-info"
    >
      Phone Number: 2
    </p>
  </div>
</DocumentFragment>
`;
